tests/core/config/test_version_migration_config.py

```diff
@@ -0,0 +1,116 @@
+# Copyright 2023 Avaiga Private Limited
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+# the License. You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+# an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations under the License.
+
+import os
+from unittest import mock
+
+import pytest
+
+from src.taipy.core._version._version_manager import _VersionManager
+from src.taipy.core.exceptions.exceptions import NonExistingVersion
+from taipy.config.config import Config
+
+
+def test_version_migration_config_creation():
+    version_migration_config = Config.set_version_migration("version_migrations1", "latest", "latest")
+
+    assert list(Config.version_migrations) == ["default", version_migration_config.id]
+
+    version_migration2_config = Config.set_version_migration("version_migrations2", "latest", "latest")
+    assert list(Config.version_migrations) == ["default", version_migration_config.id, version_migration2_config.id]
+
+
+def test_version_migration_count():
+    Config.set_version_migration("version_migrations1", "latest", "latest")
+    assert len(Config.version_migrations) == 2
+
+    Config.set_version_migration("version_migrations2", "latest", "latest")
+    assert len(Config.version_migrations) == 3
+
+    Config.set_version_migration("version_migrations3", "latest", "latest")
+    assert len(Config.version_migrations) == 4
+
+
+def test_version_migration_getitem():
+    version_migration_config_id = "version_migrations1"
+    version_migration = Config.set_version_migration(
+        version_migration_config_id,
+        target_version="latest",
```

Comment:
SHould this raise a config error ?

Reply:
It's not, because the `_VersionManager` can translate "latest" -> the latest version. 
Similar for "dev" or "development" -> the development version.

In the future, we may want to limit the target and source version to production versions only, but for now, I'll leave it as it is.

Reply:
I mean what is the meaning of migrating from version latest to version latest?